tests/aws/services/stepfunctions/v2/scenarios/test_base_scenarios.py

```diff
@@ -961,6 +961,84 @@ def test_map_item_reader_base_csv_headers_first_line(
             exec_input,
         )
 
+    @markers.aws.validated
+    @pytest.mark.parametrize(
+        "max_items_value",
+        [0, 2, 100_000_000],  # Linter on creation filters for valid input integers (0 - 100000000).
+    )
+    def test_map_item_reader_csv_max_items(
+        self,
+        aws_client,
+        s3_create_bucket,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+        max_items_value,
+    ):
+        bucket_name = s3_create_bucket()
+        sfn_snapshot.add_transformer(RegexTransformer(bucket_name, "bucket-name"))
+
+        key = "file.csv"
+        csv_file = (
+            "Col1,Col2\n" "Value1,Value2\n" "Value3,Value4\n" "Value5,Value6\n" "Value7,Value8\n"
+        )
+        aws_client.s3.put_object(Bucket=bucket_name, Key=key, Body=csv_file)
+
+        template = ST.load_sfn_template(ST.MAP_ITEM_READER_BASE_CSV_MAX_ITEMS)
+        template["States"]["MapState"]["ItemReader"]["ReaderConfig"]["MaxItems"] = max_items_value
+        definition = json.dumps(template)
+
+        exec_input = json.dumps({"Bucket": bucket_name, "Key": key})
+        create_and_record_execution(
+            aws_client.stepfunctions,
+            create_iam_role_for_sfn,
+            create_state_machine,
+            sfn_snapshot,
+            definition,
+            exec_input,
+        )
+
+    @markers.aws.validated
+    @pytest.mark.parametrize(
+        "max_items_value", [-1, 0, 2, 1.5, 100_000_000, 100_000_001]
```

Comment:
is the 2 before 1.5 intentional?

Reply:
I can definitely put them in order, this was a mild attempt to group valid inputs together but go lost during filtering the test inputs!